docs/guide/named-expressions.md

```diff
@@ -24,20 +24,42 @@ the names as follows:
 
 ## Examples
 
-| Type | Custom name | Example expression |
-| :--- | :--- | :--- |
-| Named cell | myCell | =A1 |
-| Named range of cells | myRange | =A1:D10 |
-| Named constant (number) | myNumber | =10 |
-| Named constant (string) | myText | ="One Small Step for Man" |
-| Named formula | myFormula | =SUM(A1:D10) |
+| Type                    | Custom name | Example expression        |
+|:------------------------|:------------|:--------------------------|
+| Named cell              | myCell      | =A1                       |
+| Named range of cells    | myRange     | =A1:D10                   |
+| Named constant (number) | myNumber    | =10                       |
+| Named constant (string) | myText      | ="One Small Step for Man" |
+| Named formula           | myFormula   | =SUM(A1:D10)              |
+
+## Name rules
+
+A name of an expression:
+- is case-insensitive,
+- must be unique within the scope,
+- must contain only UNICODE letters, numbers, an underscore character and a dot character,
+- must start with a letter or an underscore,
+- must be distinctive from a cell reference in A1-style, e.g. `Q4`, `YEAR2023`,
+- must be distinctive from a cell reference in R1C1-style, e.g. `R4C5`, `RC`, `R0C`.
 
-## Naming convention
+**Examples of correct and incorrect names:**
+
+| Name        | Validity  |
+|:------------|:----------|
+| my Revenue  | Incorrect |
+| myRevenue   | Correct   |
+| quarter1    | Incorrect |
+| quarter_1   | Correct   |
+| 1stQuarter  | Incorrect |
+| _1stQuarter | Correct   |
+| A1          | Incorrect |
+| $A$1        | Incorrect |
+| RC          | Incorrect |
```

Comment:
I think adding two more example with `.` would be great. First which is incorrect - starting with `.` and second which is correct and has `.` in the middle.

Reply:
Added